services/spar/src/Spar/API.hs

```diff
@@ -133,37 +140,62 @@ withDebugLog msg showval action = do
   SAML.logger SAML.Debug $ "leaving " ++ msg ++ mconcat [": " ++ fromJust mshowedval | isJust mshowedval]
   pure val
 
-authorizeIdP :: (HasCallStack, MonadError ServantErr m, Brig.MonadSparToBrig m)
+-- | Called by get, delete handlers.
+authorizeIdP :: (HasCallStack, MonadError SparError m, SAML.SP m, Brig.MonadSparToBrig m)
              => ZUsr -> IdP -> m IdP
-authorizeIdP Nothing _ = throwError err403 { errBody = "Auth token required" }
 authorizeIdP zusr idp = do
   teamid <- getZUsrTeam zusr
   if teamid == idp ^. SAML.idpExtraInfo
-    then pure idp
-    else throwError err403 { errBody = "Wrong or invalid auth token or not in a team" }
+    then idp <$ Brig.assertIsTeamOwner zusr teamid
+    else throwSpar SparNotInTeam
 
-getZUsrTeam :: (HasCallStack, MonadError ServantErr m, Brig.MonadSparToBrig m)
+-- | Called by post handler, and by 'authorizeIdP'.
+getZUsrTeam :: (HasCallStack, MonadError SparError m, SAML.SP m, Brig.MonadSparToBrig m)
             => ZUsr -> m Brig.TeamId
-getZUsrTeam Nothing = throwError err403 { errBody = Aeson.encode [aesonQQ|{"error":"no auth token"}|] }
+getZUsrTeam Nothing = throwSpar SparNotInTeam
 getZUsrTeam (Just uid) = do
   usr <- Brig.getUser uid
   case Brig.userTeam =<< usr of
-    Nothing -> throwError err403 { errBody = Aeson.encode [aesonQQ|{"error":"you need to be team admin to create an IdP"}|] }
+    Nothing -> throwSpar SparNotInTeam
     Just teamid -> pure teamid
 
-initializeIdP :: (MonadError ServantErr m, SAML.SP m) => NewIdP -> Brig.TeamId -> m IdP
+initializeIdP :: (MonadError SparError m, SAML.SP m) => NewIdP -> Brig.TeamId -> m IdP
 initializeIdP (NewIdP _idpMetadata _idpIssuer _idpRequestUri _idpPublicKey) _idpExtraInfo = do
   _idpId <- SAML.IdPId <$> SAML.createUUID
   pure SAML.IdPConfig {..}
 
-validateNewIdP :: (MonadError ServantErr m) => NewIdP -> m ()
-validateNewIdP _idp = pure ()
--- TODO:
--- [aesonQQ|{"error":"not a SAML metainfo URL"}|]
--- [aesonQQ|{"error":"invalid or unresponsive request URL"}|]
--- [aesonQQ|{"error":"public keys in request body and metainfo do not match"}|]
 
--- Type families to convert spar's 'API' type into an "outside-world-view" API type
+type MonadValidateIdP m = (MonadHttp m, MonadIO m)
+
+validateNewIdP :: forall m. (HasCallStack, MonadError SparError m, MonadValidateIdP m)
+               => NewIdP -> m ()
+validateNewIdP newidp = if True then pure () else do
```

Comment:
`if True then pure ()` ? :)

Reply:
will be removed once the remainder of this function won't crash unconditionally any more.  :-).  sorry, should have added a TODO to that end.